defaults/preferences/prefs.js

```diff
@@ -15,6 +15,8 @@
 // App-specific preferences, in alphabetical order.
 
 pref('browser.dom.window.dump.enabled', true);
+pref('browser.hiddenWindowChromeURL', 'chrome://qbrt/content/main.html');
```

Comment:
browser.hiddenWindowChromeURL isn't actually part of the shell, it's a URL that gets loaded into the "hidden window" that Gecko always opens on startup (in order to enable an app-global menubar on Mac).

This branch tries to reuse that hidden window to load the main entry script in a browsing context (i.e. a web page), so the script has access to Web APIs in addition to XPCOM APIs. The default value for the pref is `resource://gre-resources/hiddenWindow.html`, which may actually work for this purpose.

Either way, though, this is an implementation detail that shouldn't affect URLs (a.k.a. hosted apps) that are running in qbrt's shell. For local/packaged apps that aren't running in qbrt's shell, it could theoretically affect them, if they're counting on the hidden window loading the default URL. But that seems unlikely, since both the default URL and this one load a pretty basic web page.

(See http://searchfox.org/mozilla-central/source/dom/res/hiddenWindow.html for the source of the page loaded by the default URL.)

Also see #73, which proposes a way to override the value of a runtime preference.